Clean up text for setup and data exploration chunk of DE instruction

allyhawkins opened the pull request:
Here I'm starting to clean up the DE instruction notebook, working my way through each of the major chunks. This PR covers the setup through some of the initial data exploration and subsetting to the samples we are going to use for the actual DE. 

Generally I struggled with the very beginning intro and how much explanation of differential expression to go into. I think I want to save the bulk of the explanation and discussion for when we get into pseudo-bulking. 

I also made some minor modifications of the plots that we had there to hopefully help walk trainees through why we may see that cell types aren't always grouping together after integration even though we just discussed that in the integration notebook. I am worried that things are getting a little long before we even get into the DE analysis, but I also think some of the exploration is important for them to see (and fun!). 

Here's a rendered copy of the notebook: 
[04-differential_expression.nb.html.zip](https://github.com/AlexsLemonade/training-modules/files/10270064/04-differential_expression.nb.html.zip)

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,255 @@ output:
+Both the `fastmnn_PCA` and `fastmnn_UMAP` obtain the dimensionality reduction calculated post-correction and integration.
+Let's start by looking at the UMAP of our data and labeling each sample.
+
+Which reduced dimensionality slot should we use for making our UMAP plot?
```
not a slot, right? reducedDim itself is the slot. So, which reduced dim

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -171,32 +270,47 @@ library_ids <- c(
+We are particularly interested in identifying differentially expressed genes in tumor cell types between ARMS and ERMS.
+Before we do that we need to ensure that we have enough biological replicates from each group to set up our experiment. 
+It is imperative to consider good experimental design and ensure that we have enough biological replicates (at least > 3 for each group) when performing differential gene expression analysis.
+Here one sample, or group of cells collected together, from one piece of tissue is one biological replicate.
```
This sentence is a bit confusing, but I think we can probably just leave it off.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,255 @@ output:
 ```
 
-Interestingly, it looks like samples from the ARMS and ERMS subdiagnoses tend to integrate with each other rather than integrate all together. 
+Interestingly, it looks like samples from the ARMS and ERMS subdiagnoses tend to integrate with or samples of the same subdiagnosis rather than integrate all together. 
````
extra "or"? 

I might phrase this more as "subdiagnoses tend to cluster together"

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,255 @@ output:
+_A word of caution when evaluating the cell type results for this dataset: The cell types for this dataset were assigned in a two stage process.
+The first stage assigned cells as tumor or non-tumor.
+The next stage further classified tumor cells into type of tumor cell (i.e., myoblast, myocyte, or mesoderm).
+Not all samples contain tumor cells that were further classified and samples will either have all `Tumor` or a mix of `Tumor_Mesoderm`, `Tumor_Myoblast`, and `Tumor_Myocyte`._
```
```suggestion
Some samples could not be further classified, so all of their tumor cells are denoted Tumor.
The samples which could be further classified have a mix of `Tumor_Mesoderm`, `Tumor_Myoblast`, and `Tumor_Myocyte`._
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,255 @@ output:
+
+Unlike with the previous datasets we have seen where all cells of the same cell type always grouped together, this dataset shows some slightly different patterns.
+We see that there are at least 2 clear distinct groups for each cell type, 1 in each half of the UMAP.
+It looks like this is because the difference in diagnosis group accounts for differences across all cells in the sample, even those from the same type. 
```
"even those from the same type" reads as ambiguous here - cell or diagnosis?

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,255 @@ output:
+Unlike with the previous datasets we have seen where all cells of the same cell type always grouped together, this dataset shows some slightly different patterns.
+We see that there are at least 2 clear distinct groups for each cell type, 1 in each half of the UMAP.
+It looks like this is because the difference in diagnosis group accounts for differences across all cells in the sample, even those from the same type. 
+We can take advantage of another `ggplot` tool, the `facet_wrap()` function to look at two (or more, but we'll start with two) variables in the `colData` at once - the cell type and the subdiagnosis.
```
```suggestion
We can take advantage of the `facet_wrap()` function from `ggplot2` to look at two (or more, but we'll start with two) variables in the `colData` at once - the cell type and the subdiagnosis.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,255 @@ output:
                        colour_by = "celltype_broad",
                        point_size= 0.5, 
                        point_alpha = 0.4,
                        other_fields = "diagnosis_group") +
```
Add a comment here that this is needed for `scater` appeasement.
They will have already seen this in the integration notebook but a reminder can't hurt, because it's weird!

jashapiro reviewed (commented):
This looks like a good start. I think the level of exploration is good, but I did have a couple of places that I thought you might be able to tighten up a bit. 

As I said in what should be my first comment, I think I would talk a bit at the start about why we want to DE with single-cell data over bulk. If we talk about cell-type abundance affecting bulk data, we can also use that as a motivation to go back to that topic... at least as a concluding thought, even if we don't actually do it.

One question I had was about the number of samples: We start with 10, but at some point it goes down to 9, and I think I missed where that happened (or it is an artifact of two different versions). 

I also wonder if you might want to start with the annotations, and come back to plotting the UMAPs? The reason is that even the first UMAP with the individual samples might raise some questions about why it doesn't look particularly "well-integrated". There still seems to be a lot of structure between samples. We might want to start with the ARMS/ERMS plot, or comment on the fact that with tumor data we might not expect perfect integration, as every tumor is unique.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,255 @@ output:
+Similar to the UMAP, this plot shows that ARMS and ERMS share a lot of the same cell types, although it is likely that these cell types may have distinct gene expression patterns defined by the associated subdiagnosis.
+
+We also see that only 6 of these libraries have tumor cells that have been further classified into cell type. 
 3 libraries contain cells that are only classified as tumor or non-tumor, and tumor cells are not further classified and the remaining library appears to not have been assigned any cell types, as they are all labeled with `NA`.
-We will not be using these 3 libraries.
+We will continue our analysis only using the 6 libraries with fully classified cell types, removing the other 3 before we proceed with differential expression.
```
If we really want to shorten this notebook, especially the "preamble" before actually getting into DE, we could consider just starting off with those 6 libraries from the get-go? But, I do like the exploration and I think it's important to see how these kinds of analysis sausages get made. 
Maybe DE is just our longest day!

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,255 @@ output:
-```{r}
-# define libraries to keep
+The reason we want to pare down our list of samples to consider is that we want to ensure that the cell types (or subpopulations) that we are interested in are present in all samples included in our differential gene expression analysis. 
+We would want to remove any samples that do not contain our population of interest as they would have no counts to contribute to the differential expression analysis.
````
```suggestion
We want to remove any samples that do not contain our cell population(s) of interest as they have no counts to contribute to the differential expression analysis.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -171,32 +270,47 @@ library_ids <- c(
 samples_to_keep <- integrated_sce$sample %in% library_ids
 rms_sce <- integrated_sce[,samples_to_keep]
 
 # remove original integrated object 
```
say why here (for memory?)

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -171,32 +270,47 @@ library_ids <- c(
   dplyr::count(subdiagnosis)
 ```
 
+We have 3 ARMS and 3 ERMS samples which is great! 
````
"great" vs "fine"? ;)

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -171,32 +270,47 @@ library_ids <- c(
-```{r}
+```{r celltype table}
 # look at distribution of cell types across all libraries 
 as.data.frame(colData(rms_sce)) %>%
+  # count number of cells in each cell type
````
Noting that line 153 pipes `colData(rms_sce)` into `as.data.frame()`. Doesn't super matter for consistency but pointing it out!

allyhawkins commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,255 @@ output:
 data_dir <- file.path("data", "rms")
 
-# integrated file containing samples to for DE 
+# integrated file containing all libraries to use for DE analysis
```
This is leftover from when I started to convert text to use libraries instead but then we went back, I'll revert this.

allyhawkins commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,255 @@ output:
+When we look at the assay names we should see that there are 3 matrices, `counts`, `logcounts`, and `fastmnn_corrected`. 
+The `counts` and `logcounts` assay correspond to the uncorrected gene expression data that has just been merged and has NOT been integrated.
+The `fastmnn_corrected` data contains the corrected gene expression data obtained from integration. 
+For this exercise we will not be using the `fastmnn_corrected` data (more on why not once we get to setting up the differential expression), but we need to be aware that it is present and be able to distinguish it from our uncorrected data. 
```
I think we might be able to make a plot (like the violin) plot we have at the end with the corrected data instead of `logcounts` and show that you get negative gene expression values? I don't know if that's really necessary though.

allyhawkins commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,255 @@ output:
-    # simplify subdiagnosis for easy plotting and DE later
-  dplyr::mutate(cell_id = paste(sample, barcode, sep = "-"),
+  # create a new columns
+  dplyr::mutate(cell_id = paste(sample, barcode, sep = "-"), # combination of barcode and sample
+                # simplify subdiagnosis for easy plotting and DE later
                 diagnosis_group = forcats::fct_recode(
                   subdiagnosis,
                   "ARMS" = "Alveolar rhabdomyosarcoma",
                   "ERMS" = "Embryonal rhabdomyosarcoma"
                 ))
```
I needed it for creating some of the plots to evaluate DE results later on and it was helpful to have as a column in the coldata. I want to leave it in for now, but when I get to cleaning up that chunk I can play around with removing it.

allyhawkins commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,255 @@ output:
   theme(axis.text.x = element_text(angle = 90))+
+  # split by diagnosis group 
+  # only show non-NA values on x-axis
   facet_wrap(~diagnosis_group, scales = "free_x")
```
No, but there is one sample that doesn't have any cell types assigned so that sample won't show up here. It gets removed when we filter to only samples with tumor cells.

jashapiro replied:
Ah... that explains it. We should probably explicitly note that somewhere

allyhawkins commented:
Thank you both for your very helpful comments and feedback! After going through all of them I decided to do a bit of reorganization that I think makes things have some more flow and also shortened it a little bit. 

- I updated the intro to include some information about RMS, including ARMS vs. ERMS biology, why they might be different and why we are interested in looking at the difference between samples from these two RMS subtypes. 
- I also changed my language throughout, referring to ARMS and ERMS as RMS subtypes as that is what they are in the literature and it was getting confusing the other way around. Let me know if it's unclear at any point cell type vs. RMS subtype. 
- I changed the order of the UMAPs, starting with the RMS subtype UMAP first and then showing cell types and the stacked bar plot before subsetting the SCE object. I also removed the UMAP by library ID. After reorganizing I wasn't really sure where to put it and if we really needed it anymore, but I can add it back if others think its necessary. 
- I removed some of the extra chunks at the bottom where we counted samples and cell types since the stacked bar plot shows all of that pretty nicely, especially now as a percentage. 
- One small note about the bar plot, the `space=free_x` option no longer exists so I couldn't make the bars the same width, so if others know of a way to do that please let me know and I can make that change. 

Attached is the most up to date version of the notebook: 
[04-differential_expression.nb.html.zip](https://github.com/AlexsLemonade/training-modules/files/10278591/04-differential_expression.nb.html.zip)

jashapiro commented:
> * One small note about the bar plot, the `space=free_x` option no longer exists

WHAT‽ There must be something to replace it.. I need to investigate this removal!

jashapiro commented:
> > * One small note about the bar plot, the `space=free_x` option no longer exists
> 
> WHAT‽ There must be something to replace it.. I need to investigate this removal!

Oh, you know what, it was never there.... it is only in `facet_grid()`, which makes some sense. So we could switch to using that, maybe.

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,259 @@ output:
+  theme(axis.text.x = element_text(angle = 90, vjust = 0.5))+
+  # split by diagnosis group 
+  # only show non-NA values on x-axis
+  facet_wrap(~diagnosis_group, 
+             scales = "free_x")
```
Using the new format for ggplot faceting (`vars()`), this might work:
```suggestion
  facet_grid(cols = vars(diagnosis_group), 
             scales = "free_x",
             space = "free_x")
```

Alternatively, the old format:
```suggestion
  facet_grid(. ~ diagnosis_group, 
             scales = "free_x",
             space = "free_x")
```

(Should we change all faceting functions to use `vars()`?)

allyhawkins replied:
We could? Or we could just use it here because we want the even spaced bars and show them `facet_grid` and `facet_wrap`. I don't feel strongly either way?

jashapiro replied:
I think if we change to use `vars()` with faceting here, we should probably do it everywhere. It is the new recommended way, so we should probably convert. (This comes also with the knowledge that `purrr` is officially moving away from `~` syntax too. I happen to think this is all a great loss, but it seems to be where we are.)

sjspielman replied:
Oh excellent, I am just writing this exact comment about whether we want to use`vars()` or `~`, so I'll not comment it again but keep the convo here.
I'm in `vars()` camp for better or worse to adhere to new style, but I am also sad about moving away from `~` more generally.

sjspielman commented on components/dictionary.txt:
```diff
@@ -341,8 +342,10 @@ scater
+Sebire
 Sergushichev
 setosa
+Shern
```
Did I sing the alphabet song to confirm that `h` really is before `i`? Yes, yes I did. It is.

allyhawkins replied:
I also sang the alphabet song and do every time I add something to the dictionary.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+- Use pseudo-bulking to prepare single-cell RNA-sequencing libraries for differential expression
 - Perform differential expression with the `DESeq2` package
-- Compare cell type abundance across samples
+- Use `ggplot` and `EnhancedVolcano` to visualize gene expression changes across cell types and samples
```
```suggestion
- Use `ggplot2` and `EnhancedVolcano` to visualize gene expression changes across cell types and samples
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
-We will continue working with samples from the [`SCPCP000005` project](https://scpca.alexslemonade.org/projects/SCPCP000005) we worked with in the integration notebook.
-This particular dataset contains 10 different samples that have been integrated using `fastMNN`, following the same procedure we outlined in `03-dataset_integration.Rmd`.
+Just like bulk RNA-seq, it is likely that one of the goals when performing single-cell RNA-seq will be to compare the gene expression of multiple samples to each other.
+Unlike in bulk RNA-seq analysis, single-cell RNA-seq analysis allows us to identify and annotate cell types or subpopulations of cells present in each of our samples.
```
either have "in" in the last and this sentence or neither
```suggestion
Unlike bulk RNA-seq analysis, single-cell RNA-seq analysis allows us to identify and annotate cell types or subpopulations of cells present in each of our samples.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+
+Prior to integration all 10 samples went through the same filtering, normalization, and dimensionality reduction.
+These 10 samples were then merged into one `SingleCellExperiment` object following the same steps outlined in `03-dataset_integration.Rmd`.
+The merged object was then integrated to obtain a corrected gene expression assay and corrected reduced dimensionality results.
```
```suggestion
The merged object was then integrated with `fastMNN` to obtain a corrected gene expression assay and corrected reduced dimensionality results.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,260 @@ output:
-```{r}
-sample_metadata
+In the `reducedDim` slots you should see `PCA` and `UMAP` which both correspond to the pre-integrated dimensionality reduction results.
+You should also see `fastmnn_PCA` and `fastmnn_UMAP` reduced dimensions, which correspond to the integrated results.
````
Noting that my integration notebook is using (caps) `fastMNN_PCA` and `_UMAP`. 
Should I sync up w/ here?

allyhawkins replied:
I think up to you on which one of us should switch? For some reason I really thought you were using all lower case but there is so much to keep track of.

jashapiro replied:
Weghing in in case this is not resolved, I would vote for `fastmnn_PCA`. I don't know why, except that I generally dislike mixed case for some reason.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+Just like in the integration notebook, this dataset also contains the cell type annotations found in the `celltype_fine` and `celltype_broad` columns of the `colData`.
+These cell types were originally assigned in [Patel _et al._ (2022)](https://doi.org/10.1016/j.devcel.2022.04.003).
+We will use these cell type assignments to set up the DE analysis below, but they are not required.
+It's important to note that DE analysis can be applied to any sub population of interest that is shared across samples and does not need to be explicitly defined by a cell type.
```
Note to check "sub population" vs "subpopulation" throughout notebook.

allyhawkins replied:
I think I've been using `subpopulation` more frequently so I'm going to switch to that.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+We will use these cell type assignments to set up the DE analysis below, but they are not required.
+It's important to note that DE analysis can be applied to any sub population of interest that is shared across samples and does not need to be explicitly defined by a cell type.
+
+Because we are going to be doing DE analysis between ARMS and ERMS samples, let's start by first looking at our integrated dataset and labeling our cells by which RMS subtype they are from.
```
```suggestion
Because we are going to be doing DE analysis between ARMS and ERMS samples, let's start by labeling cells in the integrated dataset based on their RMS subtype.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
-    # simplify subdiagnosis for easy plotting and DE later
-  dplyr::mutate(cell_id = paste(sample, barcode, sep = "-"),
+  # create a new columns
+  dplyr::mutate(cell_id = paste(sample, barcode, sep = "-"), # combination of barcode and sample
```
We might `glue::glue()` here instead as is done in integration

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,260 @@ output:
 ```
 
-Interestingly, it looks like samples from the ARMS and ERMS subdiagnoses tend to integrate with each other rather than integrate all together. 
+Interestingly, it looks like samples from the ARMS and ERMS subtypes tend to cluster with samples of the same subtype rather than cluster all together. 
````
Note that this is not a _technical_ "cluster."

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
-*Talk about looking across cell types and comparing cell types* 
+A word of caution when evaluating the cell type results for this dataset: The cell types for this dataset were assigned in a two stage process as described in [Patel _et al._ (2022)](https://doi.org/10.1016/j.devcel.2022.04.003).
+The first stage assigned cells as tumor or non-tumor.
+The next stage further classified tumor cells into type of tumor cell (i.e., myoblast, myocyte, or mesoderm).
```
```suggestion
The next stage further classified tumor cells into one of three types of tumor cells: myoblast, myocyte, or mesoderm.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,260 @@ output:
+                       point_alpha = 0.4)
+```
+
+That doesn't appear to be the case with this dataset. 
````
Since we detoured to talk about tumor cell types I had forgotten by now what "that" meant! Can we just re-contextualize this sentence a bit?

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+
+That doesn't appear to be the case with this dataset. 
+Unlike with the previous datasets we have seen where all cells of the same cell type always grouped together, this dataset shows some slightly different patterns and not all cells of the same cell type cluster together.
+One reason is that tumor data can be heterogeneous and every tumor is unique.
```
❄️

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+That doesn't appear to be the case with this dataset. 
+Unlike with the previous datasets we have seen where all cells of the same cell type always grouped together, this dataset shows some slightly different patterns and not all cells of the same cell type cluster together.
+One reason is that tumor data can be heterogeneous and every tumor is unique.
+Depending on the tumor type we may not expect every sample to integrate perfectly.
```
I'd be a bit more explicit - tumor types that are more heterogeneous will be harder to integrate together.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+Unlike with the previous datasets we have seen where all cells of the same cell type always grouped together, this dataset shows some slightly different patterns and not all cells of the same cell type cluster together.
+One reason is that tumor data can be heterogeneous and every tumor is unique.
+Depending on the tumor type we may not expect every sample to integrate perfectly.
+In this particular case we are looking at two subtypes of RMS that have been shown to have distinctly different mutation burdens and differentiation states, so it's likely that those differences are contributing to how well they will be integrated.
```
```suggestion
In this particular case we are looking at two subtypes of RMS that have distinctl mutation burdens and differentiation states, so it's likely that those differences contribute to how well they integrate.
``

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+Depending on the tumor type we may not expect every sample to integrate perfectly.
+In this particular case we are looking at two subtypes of RMS that have been shown to have distinctly different mutation burdens and differentiation states, so it's likely that those differences are contributing to how well they will be integrated.
+
+To confirm our hypothesis that cells are grouping together both by cell type and by RMS subtype we can create a plot that incorporates both pieces of metadata.
```
I would say this more of an exploration of integration success than hypothesis, especially since we have "real" hypothesis coming up with DE testing. What do you think?

```suggestion
To explore whether cells are grouping together both by cell type and by RMS subtype, we can create a plot that incorporates both pieces of metadata.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+In this particular case we are looking at two subtypes of RMS that have been shown to have distinctly different mutation burdens and differentiation states, so it's likely that those differences are contributing to how well they will be integrated.
+
+To confirm our hypothesis that cells are grouping together both by cell type and by RMS subtype we can create a plot that incorporates both pieces of metadata.
+We will take advantage of the `facet_wrap()` function from `ggplot2` to look at two (or more, but we'll start with two) variables in the `colData` at once - the cell type and the subdiagnosis.
```
wait..this seems to imply we could have 3-4 facet variables, but that's too many plot dimensions for human brains.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+
+To confirm our hypothesis that cells are grouping together both by cell type and by RMS subtype we can create a plot that incorporates both pieces of metadata.
+We will take advantage of the `facet_wrap()` function from `ggplot2` to look at two (or more, but we'll start with two) variables in the `colData` at once - the cell type and the subdiagnosis.
+In the below plot we will color our cells by cell type but we will also use `facet_wrap()` to plot the cells from each subdiagnosis in their own separate panel of the plot.
```
```suggestion
In the below plot we will color our cells by cell type while also using `facet_wrap()` so that cells from different subdiagnoses will be in their own plot panel.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,260 @@ output:
+We can also use a stacked barplot to look at the distribution of cell types across each sample.
+
+```{r celltype barplot}
+# filter coldata data.frame to only include tumor cells
````
Is this our first time (or one of very few times!) using `stringr`? I'd add a comment into the pipe about `stringr::str_detect()` arguments and it returns T/F.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+    geom_bar(position = "fill", color = "black", size = 0.2) +
     labs(
-      x = "Sample",
+      x = "Library",
```
sample/library?

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+We will continue our analysis only using the 6 libraries with fully classified cell types, removing the other 3 before we proceed with differential expression.
+
+The reason we want to pare down our list of samples to consider is that we want to ensure that the cell types (or subpopulations) that we are interested in are present in all samples included in our DE analysis. 
+We want to remove any samples that do not contain our cell population(s) of interest as they have no counts to contribute to the DE analysis.
```
There's a bit of nuance here - those 3 libraries probably do have those (or at least some) of those cell types, but we don't know which ones those are because of classification issues.

allyhawkins replied:
This is true, but I don't think I want to get too far into the weeds of the classifications here.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -171,32 +275,26 @@ library_ids <- c(
 rms_sce <- integrated_sce[,samples_to_keep]
 
-# remove original integrated object 
+# remove original integrated object to save some memory
```
Oh boy I read this as save money (even though this was originally my comment) and I was like, "wow, does this really affect server cost!? I'm surprised!"
Having fun!

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -171,32 +275,26 @@ library_ids <- c(
-The `Tumor_Myoblast` is the most prominent cell type, and based on the UMAPs is found in all 6 of our samples.
-We will start by comparing the gene expression of the myoblasts across ARMS and ERMS samples.
+We now have an updated SCE object that contains 6 samples that were obtained from a mix of ARMS and ERMS patients.
+We are particularly interested in identifying DE genes in tumor cell types between ARMS and ERMS.
```
I'd spell out DE here since it's got some grammar vs DE which I tend to read as "differential expression" (but is that just me?)
```suggestion
We are particularly interested in identifying differentially expressed genes for specific tumor cell types between ARMS and ERMS.
```

sjspielman replied:
I'll note one could phrase this as a question, something like: "We want to ask, Are specific tumor cell types differentially expressed between ARMS and ERMS?"

Ok either way!

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -171,32 +275,26 @@ library_ids <- c(
+We now have an updated SCE object that contains 6 samples that were obtained from a mix of ARMS and ERMS patients.
+We are particularly interested in identifying DE genes in tumor cell types between ARMS and ERMS.
+We should also make sure that we have enough biological replicates from each group to set up our experiment. 
+It is imperative to consider good experimental design and ensure that we have enough biological replicates (at least > 3 for each group) when performing differential gene expression analysis.
```
at least 3, not at least >3!
although we can dream..

```suggestion
It is imperative to consider good experimental design and ensure that we have enough biological replicates (at least 3 for each group) when performing differential gene expression analysis.
```

sjspielman reviewed (commented):
I think this is coming together very nicely! I do like how the introduction has come together here! It’s a good blend of technical background with the biology of the actual analysis which I think helps to really contextualize the technicality.

allyhawkins commented:
I made a few more smaller edits including switching over to `facet_grid` and using `vars` if people would like to take another look.

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+This particular dataset contains 10 different samples that have been integrated using `fastMNN`, following the same procedure we outlined in `03-dataset_integration.Rmd`.
+These 10 samples represent two different types of rhabdomyosarcoma (RMS): embryonal rhabdomyosarcoma (ERMS) and alveolar rhabdomyosarcoma (ARMS).
+These two subtypes can be distinguished by the presence of the `PAX3/PAX7-FOXO1` fusion gene present only in ARMS patients.
+Additionally, cells found in ARMS tumors tend to have an increased mutational burden and cells are in a more differentiated state in comparison to ERMS tumor cells ([Shern _et al._ 2014](https://doi.org/10.1158/2159-8290.CD-13-0639), [Stewart _et al._](https://doi.org/10.1016/j.ccell.2018.07.012)).
```
```suggestion
Additionally, cells found in ARMS tumors tend to have an increased mutational burden and cells are in a more differentiated state in comparison to ERMS tumor cells ([Shern _et al._ 2014](https://doi.org/10.1158/2159-8290.CD-13-0639); [Stewart _et al._ 2018](https://doi.org/10.1016/j.ccell.2018.07.012)).
```

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
+These two subtypes can be distinguished by the presence of the `PAX3/PAX7-FOXO1` fusion gene present only in ARMS patients.
+Additionally, cells found in ARMS tumors tend to have an increased mutational burden and cells are in a more differentiated state in comparison to ERMS tumor cells ([Shern _et al._ 2014](https://doi.org/10.1158/2159-8290.CD-13-0639), [Stewart _et al._](https://doi.org/10.1016/j.ccell.2018.07.012)).
+RMS tumors, regardless of subtype, are made up of cells typically associated with development of skeletal muscle: mesoderm, myoblasts, and myocytes ([Sebire and Malone 2003](https://doi.org/10.1136/jcp.56.6.412)).
+The group that published the data we are going to look at in this notebook identified distinct differences between ARMS and ERMS tumors. 
```
This sentence seems a bit vague. I don't know if we want to be more specific, or maybe just combine this sentence with the next one?

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
-library(SingleCellExperiment)
-library(ggplot2)
+# load libraries
+library(magrittr) # the pipe (%>%)
```
dunno if you want to do the pipe substitution for this PR...

allyhawkins replied:
I actually had meant to do that in the next PR and do a replace all. I'll add it in there instead.

sjspielman replied:
Just noting that this is happening in #600 instead

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
 integrated_sce_file <- file.path(data_dir, "integrated", "rms_all_sce.rds")
 
-# sample metadata with information needed to setup DE
+# sample metadata with information needed to setup DE analysis
```
These comments may be a bit long. I'm going to file an issue to check line length in the automatically rendered versions of these notebooks, as that can be an issue.
```suggestion
# sample metadata with information needed to set up DE analysis
```

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,260 @@ output:
+Here we should have both the original (uncorrected) data and the integrated (corrected) data for both the gene expression and the reduced dimensionality results.
+How are those stored in our object?
+
+```{r print sce}
+# print out entire object
+integrated_sce
+
+# look at the assay names in our object
+assayNames(integrated_sce)
+
+# look at the names of the dimensionality reduction present in the SCE object
+reducedDimNames(integrated_sce)
 ```
````
These shoudl probably be separate chunks, to keep the output separate when running interactively. When rendered, the chunk will be broken by the outputs, but in an interactive session, all three will appear in the same place.

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,260 @@ output:
+```{r modify coldata}
+# add the sample metadata to the colData from the integrated SCE object
 coldata_df <- colData(integrated_sce) %>%
+  # convert from DataFrame to data.frame
````
We will have done this a few time by this point, so we might not need this comment, but it is fine to leave it in!

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,260 @@ output:
 
-# add back to sce 
+# add modified data.frame back to SCE as DataFrame
 colData(integrated_sce) <- DataFrame(coldata_df, row.names = coldata_df$cell_id)
```
I anticipate this may push the line limit length for rendering, which may be < 80 characters
```suggestion
colData(integrated_sce) <- DataFrame(coldata_df, 
                                     row.names = coldata_df$cell_id)
```

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -171,32 +275,27 @@ library_ids <- c(
-sample_metadata %>%
-  dplyr::filter(library_id %in% library_ids) %>%
-  dplyr::count(subdiagnosis)
+# print out our new sce 
```
I have been capitalizing SCE when not in a variable name or similar. I just noticed this one, but we probably want to be consistent.
```suggestion
# print out our new SCE 
```

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -171,32 +275,27 @@ library_ids <- c(
 rms_sce <- integrated_sce[,samples_to_keep]
 
-# remove original integrated object 
+# remove original integrated object to save some memory
```
For a modicum of "safety" I would move the removal to _after_ printing out the new SCE. Always check what you have before removing something that might take some time to reload/reproduce!

"Now that we know we have what we want, we can remove the old version to save memory."

jashapiro commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -171,32 +275,27 @@ library_ids <- c(
 
+If we look back at our stacked barplot we see that we picked 3 ARMS and 3 ERMS samples.
+We can also see that the majority of cells are tumor cells, in particular the largest population of cells appears to be the `Tumor_Myoblast`. 
+For this example we will focus on identifying DE genes in one cell type, the `Tumor_Myoblast` cells, but the principles applied below can be applied to any cell types or subpopulations of interest.
```
Just tightening up a bit.
```suggestion
For this example we will focus on identifying DE genes in these `Tumor_Myoblast` cells, but the principles applied below can be applied to any cell types or subpopulations of interest.
```

jashapiro reviewed (approved):
This looks good to me!

I had a few minor comments about things like line lengths, chunk splitting, and when we remove an object, but nothing that requires another detailed look at this stage.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
+Just like bulk RNA-seq, it is likely that one of the goals when performing single-cell RNA-seq will be to compare the gene expression of multiple samples to each other.
+Unlike bulk RNA-seq analysis, single-cell RNA-seq analysis allows us to identify and annotate cell types or subpopulations of cells present in each of our samples.
+This means that we can account for differences in cell type composition across samples and specifically focus on cell types or populations of interest when performing differential expression (DE) analysis.
+In this notebook, we will work with multiple samples to identify differentially expressed genes across cell types of interest using the [`DESeq2`](https://bioconductor.org/packages/release/bioc/html/DESeq2.html) package. 
```
Would we prefer to tag this version?


```suggestion
In this notebook, we will work with multiple samples to identify differentially expressed genes across cell types of interest using the [`DESeq2`](https://bioconductor.org/packages/3.16/bioc/html/DESeq2.html) package. 
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
+This particular dataset contains 10 different samples that have been integrated using `fastMNN`, following the same procedure we outlined in `03-dataset_integration.Rmd`.
+These 10 samples represent two different types of rhabdomyosarcoma (RMS): embryonal rhabdomyosarcoma (ERMS) and alveolar rhabdomyosarcoma (ARMS).
+These two subtypes are distinguished by the presence of the `PAX3/PAX7-FOXO1` fusion gene, which is present only in ARMS patients.
+Additionally, cells found in ARMS tumors tend to have an increased mutational burden and cells are in a more differentiated state in comparison to ERMS tumor cells ([Shern _et al._ 2014](https://doi.org/10.1158/2159-8290.CD-13-0639); [Stewart _et al._ 2018](https://doi.org/10.1016/j.ccell.2018.07.012)).
```
```suggestion
Additionally, cells found in ARMS tumors tend to have an increased mutational burden with cells in a more differentiated state compared to ERMS tumor cells ([Shern _et al._ 2014](https://doi.org/10.1158/2159-8290.CD-13-0639); [Stewart _et al._ 2018](https://doi.org/10.1016/j.ccell.2018.07.012)).
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
+These two subtypes are distinguished by the presence of the `PAX3/PAX7-FOXO1` fusion gene, which is present only in ARMS patients.
+Additionally, cells found in ARMS tumors tend to have an increased mutational burden and cells are in a more differentiated state in comparison to ERMS tumor cells ([Shern _et al._ 2014](https://doi.org/10.1158/2159-8290.CD-13-0639); [Stewart _et al._ 2018](https://doi.org/10.1016/j.ccell.2018.07.012)).
+RMS tumors, regardless of subtype, are made up of cells typically associated with development of skeletal muscle: mesoderm, myoblasts, and myocytes ([Sebire and Malone 2003](https://doi.org/10.1136/jcp.56.6.412)).
+In [Patel _et al._ (2022)](https://doi.org/10.1016/j.devcel.2022.04.003), they tested the hypothesis that cell types have distinct gene expression patterns in ARMS vs. ERMS samples.
```
```suggestion
[Patel _et al._ (2022)](https://doi.org/10.1016/j.devcel.2022.04.003) tested the hypothesis that cell types have distinct gene expression patterns in ARMS vs. ERMS samples.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,267 @@ output:
-```{r}
+We will start by reading in a `SingleCellExperiment` (SCE) object that contains both the uncorrected (merged but not integrated) and corrected (integrated) gene expression data for all 10 samples.
+
+Prior to integration all 10 samples went through the same filtering, normalization, and dimensionality reduction.
````
```suggestion
Prior to integration all 10 samples, went through the same filtering, normalization, and dimensionality reduction.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
+The final SCE object was stored in `data/rms/integrated/rms_all_sce.rds`.
+
+We also have provided a metadata file, `data/rms/annotations/rms_sample_metadata.tsv`, that contains information from each sample, such as diagnosis, sex, age, etc.
+In this file each row corresponds to a sample found in the integrated SCE object.
```
This somehow sounds more right to my ear, but doesn't matter!

```suggestion
Each row in this life corresponds to a sample found in the integrated SCE object.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
+
+Just like in the integration notebook, this dataset also contains the cell type annotations found in the `celltype_fine` and `celltype_broad` columns of the `colData`.
+These cell types were originally assigned in [Patel _et al._ (2022)](https://doi.org/10.1016/j.devcel.2022.04.003).
+We will use these cell type assignments to set up the DE analysis below, but they are not required.
```
```suggestion
We will use these cell type assignments to set up the DE analysis below, but they are not required for DE analysis itself.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
+Just like in the integration notebook, this dataset also contains the cell type annotations found in the `celltype_fine` and `celltype_broad` columns of the `colData`.
+These cell types were originally assigned in [Patel _et al._ (2022)](https://doi.org/10.1016/j.devcel.2022.04.003).
+We will use these cell type assignments to set up the DE analysis below, but they are not required.
+It's important to note that DE analysis can be applied to any subpopulation of interest that is shared across samples and does not need to be explicitly defined by a cell type.
```
```suggestion
It's important to note that DE analysis can be applied to any subpopulation of interest that is shared across samples besides just cell types.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
+We will use these cell type assignments to set up the DE analysis below, but they are not required.
+It's important to note that DE analysis can be applied to any subpopulation of interest that is shared across samples and does not need to be explicitly defined by a cell type.
+
+Because we are going to be doing DE analysis between ARMS and ERMS samples, let's start by labeling cells in the integrated dataset based on their RMS subtype.
```
```suggestion
Because we'll performg DE analysis between ARMS and ERMS samples, let's start by labeling cells in the integrated dataset based on their RMS subtype.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,267 @@ output:
+```
+
+Looking at this sample table, we see a column named `subdiagnosis` which accounts for the RMS subtype, ARMS or ERMS.
+We also see other columns that contain information about each specific sample and may contribute to differences seen across the samples.
````
"and may contribute to differences seen across the samples" seems like overkill here, or we would want to actually explore it?

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,267 @@ output:
+# look at the head of the coldata
+head(colData(integrated_sce))
+```
+
````
We might want a transition sentence here like "uh oh, not there, where can we find it?"

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
-    # simplify subdiagnosis for easy plotting and DE later
-  dplyr::mutate(cell_id = paste(sample, barcode, sep = "-"),
+  # create a new columns
+  dplyr::mutate(cell_id = glue::glue("{sample}-{barcode}"), # combination of barcode and sample
```
check line length here -  may get weirdly wrapped?

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,267 @@ output:
-We can also look at the distribution of cell types across each sample.
 
-```{r}
+As expected, we see that the visual clusters of cell types are being separated, most likely due to different RMS subtypes.
````
```suggestion
As expected, we see that cell types are separated, most likely due to different RMS subtypes.
```

sjspielman replied:
some phrasing was funky here, but you don't need to use my suggestion exactly.

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,267 @@ output:
-```{r}
+As expected, we see that the visual clusters of cell types are being separated, most likely due to different RMS subtypes.
+
+We can also use a stacked barplot to look at the distribution of cell types across each sample.
````
```suggestion
We can also use a stacked barplot to look at the distribution of cell types across each sample, which will require a bit of wrangling first.
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,267 @@ output:
+We can also use a stacked barplot to look at the distribution of cell types across each sample.
+
+```{r celltype barplot}
+# filter coldata data.frame to only include tumor cells
````
```suggestion
# filter the coldata data frame to only include tumor cells
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
-  theme(axis.text.x = element_text(angle = 90))+
-  facet_wrap(~diagnosis_group, scales = "free_x")
+  theme(axis.text.x = element_text(angle = 90, vjust = 0.5))+
+  # split by diagnosis group 
```
```suggestion
  # facet by diagnosis group 
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
-  facet_wrap(~diagnosis_group, scales = "free_x")
+  theme(axis.text.x = element_text(angle = 90, vjust = 0.5))+
+  # split by diagnosis group 
+  # only show non-NA values on x-axis
```
which argument accomplishes this?

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
````diff
@@ -12,156 +12,267 @@ output:
-```{r}
-# define libraries to keep
+We also see that only 6 of these libraries have tumor cells that have been further classified into mesoderm, myoblast, and myocyte. 
+3 libraries contain cells that are only classified as tumor or non-tumor, and tumor cells are not further classified, and the remaining library appears to not have been assigned any cell types and is not present in our plot, as they are all labeled with `NA`.
````
```suggestion
3 libraries contain cells that are only classified as tumor or non-tumor, and tumor cells are not further classified, and the remaining library is not even present in our plot because it was not assigned any cell types (all are `NA`).
```

sjspielman commented on scRNA-seq-advanced/04-differential_expression.Rmd:
```diff
@@ -12,156 +12,267 @@ output:
-# define libraries to keep
+We also see that only 6 of these libraries have tumor cells that have been further classified into mesoderm, myoblast, and myocyte. 
+3 libraries contain cells that are only classified as tumor or non-tumor, and tumor cells are not further classified, and the remaining library appears to not have been assigned any cell types and is not present in our plot, as they are all labeled with `NA`.
+We will continue our analysis only using the 6 libraries with fully classified cell types, removing the other 3 before we proceed with differential expression.
```
```suggestion
We will continue our analysis only using the 6 libraries with fully classified cell types, removing the other 4 before we proceed with differential expression.
```

sjspielman reviewed (approved):
🚀 Some small comments, don't need to see again! Take or leave my wording changes.
